docs/pages/blog/turbo-1-2-0.mdx

```diff
@@ -0,0 +1,112 @@
+
+- [**New Task Filtering API**](#new-task-filtering-api): `--filter` flag adds more expressive and intuitive task filtering capabilities to `turbo run`
+- [**Human-readable and JSON dry runs**](#debug-and-automate-with---dry-run): `--dry-run` flag can print out information about a `turbo run` without executing any tasks, in both human and JSON-parse friendly formats
+- [**Improved Internal Scheduler and Graph**](#improved-internal-scheduler-and-graph): We refactored `turbo` 's internal scheduler and graph to be more ergonomic and predictable
+- [**Enhanced Remote Cache Security**](#cache-outputs-integrity-and-signature-verification): Cryptographically sign remote cache artifacts with your own secret key
+
+Update today by running `npm install turbo@latest`. After running `turbo run` for the first time, you'll see instructions about how to use `@turbo/codemod` to run automatic migrations for `v1.2`.
+
+## New Task Filtering API
+
+We are excited to release one of our most requested features: the ability to expressively filter tasks through a `--filter` flag. The `--filter` flag is the much more powerful successor to the current combination of `--scope`, `--include-dependencies`, `--since`, and `--no-deps` flags. With `--filter` you can tell `turbo` to restrict executing commands to a subset of matched packages in your monorepo based on name, folder, and even if it has changed since a git commit ref.
+
+Take a look at some examples of what you can accomplish with the new `--filter` command:
+
+- `--filter=<package_name>` - match by exact package name or glob pattern
+- `--filter=...<package_name>`- match by package name/glob and include all dependent packages of match(es)
+- `--filter=...^<package_name>`- match by package name/glob and include all dependent packages of matches, but exclude the matches themselves
+- `--filter=<package_name>...` - match by package name/glob and include all the matched packages' dependencies
+- `--filter=<package_name>^...` - match by package name/glob and include all matched package dependencies, but exclude the matches themselves
+- `--filter={./path/to/package}` - match by path or filesystem glob pattern
+- `--filter=[origin/main]` - match by changed packages since a git commit ref
+
+You can use multiple filters together to get even more granular filtering as well as combine each part of the above patterns `{}`, `[]` , `^` , and `...` to express more complex behavior.
+
+For example, assuming you have an app located in `./apps/web` directory, some local packages that it uses as dependencies, and a Turborepo pipeline where `test` depends on `^build` [topologically](https://turborepo.org/docs/glossary#topological-order), running:
+
+```sh
+turbo run test --filter={./apps/web}[HEAD^1]^...
+```
+
+would tell `turbo` to ensure that dependencies are built and to run the `test` script in all of the local dependencies of the app located in `./apps/web`, not including that app itself, if the app has changed since HEAD^1.
```

Comment:
```suggestion
would tell `turbo` to ensure dependencies are built and to run the `test` script in all of the local dependencies of the app located in `./apps/web`, not including that app itself, if the app has changed since HEAD^1.
```